File: 1284-four-divisors/four-divisors.py
class Solution(object):
    def sumFourDivisors(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ans = 0

        for x in nums:
            cnt = 0
            sum = 0
            d=1
            while(d*d<=x):
                if (x % d == 0):
                    if (d * d == x):
                        cnt += 1
                        sum += d
                    else :
                        cnt += 2
                        sum += d + x / d
                    
                    if (cnt > 4): break  
                d+=1 
                
            

            if (cnt == 4):
                ans += sum
            
        

        return ans

File: 1284-four-divisors/four-divisors.cpp
class Solution {
public:
    int sumFourDivisors(vector<int>& nums) {
        int ans = 0;

        for (int x : nums) {
            int cnt = 0, sum = 0;

            for (int d = 1; d * d <= x; d++) {
                if (x % d == 0) {
                    if (d * d == x) {
                        cnt += 1;
                        sum += d;
                    } else {
                        cnt += 2;
                        sum += d + x / d;
                    }
                    if (cnt > 4) break;   // early exit
                }
            }

            if (cnt == 4) {
                ans += sum;
            }
        }

        return ans;
    }
};